src/closure.jl

```diff
@@ -286,15 +288,30 @@ function enclose(exorig::Expr, reserve_per, minbatchsize, per::Symbol, mod)
     # end
   end
   closure = Symbol("##closure##")
+
+  threadlocal_init_single   = threadlocal == :() ? :() : :( $threadlocal_var = $threadlocal )
+  threadlocal_repack_single = threadlocal == :() ? :() : :( threadlocal = [threadlocal] )
+  threadlocal_init1         = threadlocal == :() ? :() : :( $threadlocal_var = Vector(undef, 0) )
+  threadlocal_init2         = threadlocal == :() ? :() : quote
+    resize!($(esc(threadlocal_var)),max(1,$(threadtup.args[2]))) # Why is the max necessary?
+    for i in eachindex($(esc(threadlocal_var)))
+      $(esc(threadlocal_var))[i] = $(esc(threadlocal))
+    end
+  end 
+  threadlocal_get           = threadlocal == :() ? :() : :( $threadlocal_var_gen = $threadlocal_var[Threads.threadid()] )
+  threadlocal_set           = threadlocal == :() ? :() : :( $threadlocal_var[Threads.threadid()] = $threadlocal_var_gen )
```

Comment:
There is a bug here, both in the get and in the set, because the `threadid()` can be bigger than the number of threads actually used, e.g. when the we have a max of 5 threads, but we use only 3 of them, with ids 1,4,5.

Reply:
This is why it may be better to edit `batch`, to try and find a way to use the `UnsignedIteratorEarlyStop` to see which threads:
```julia
julia> using PolyesterWeave

julia> PolyesterWeave.request_threads(5)
((Thread (5) Iterator: U[1, 2, 3, 4, 5],), (0x000000000000001f,))

julia> PolyesterWeave.request_threads(5)
((Thread (5) Iterator: U[6, 7, 8, 9, 10],), (0x00000000000003e0,))
```